pip/req/req_set.py

```diff
@@ -175,7 +175,11 @@ def locate_files(self):
                                 )):
                             req_to_install.conflicts_with = \
                                 req_to_install.satisfied_by
-                        req_to_install.satisfied_by = None
+                        # don't install if this exact requirement is already
+                        # installed
+                        if not (self.satisfied_by.version ==
```

Comment:
`self.satisfied_by` does not seem to be defined, I guess this should be `req_to_install.satisfied_by.version`
and yes, this needs tests ;)